Dockerfile

```diff
@@ -8,16 +8,27 @@ WORKDIR /app
 
 COPY pyproject.toml poetry.lock ./
 COPY . .
-RUN poetry install
+#RUN poetry install
 
 EXPOSE 5100
 
 COPY /todo_app/ ./todo_app/
 
 
-FROM base as production
-ENTRYPOINT [ "poetry", "run", "gunicorn", "--workers=2", "--bind", "0.0.0.0:80", "todo_app.app:create_app()" ]
+# dev stage
+FROM base as development
+RUN poetry install
+ENTRYPOINT [ "poetry", "run", "flask", "run", "--host", "0.0.0.0", "--port", "80"]
 
 
-FROM base as development
-ENTRYPOINT [ "poetry", "run", "flask", "run", "--host", "0.0.0.0", "--port", "80"]
\ No newline at end of file
+# testing stage
+FROM base as test
+RUN poetry install
+ENTRYPOINT ["poetry", "run", "pytest"]
+
+
+# production stage
+FROM base as production
+RUN poetry config virtualenvs.create false --local && poetry install
+#ENTRYPOINT [ "poetry", "run", "gunicorn", "--workers=2", "--bind", "0.0.0.0:80", "todo_app.app:create_app()" ]
+CMD poetry run gunicorn "todo_app.app:create_app()" --bind 0.0.0.0:$PORT
```

Comment:
Now that you've moved to an environment variable for the PORT, you'll likely find your old instructions for running the production container don't work, since they don't set this variable. A nice fix for that can be to add a default value (e.g. with `ENV PORT=80`) inside this production build stage